content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java

```diff
@@ -4250,6 +4250,41 @@ public int compare(Group g0, Group g1) {
 					// something failed, group filter will be hidden
 				}
 			}
+
+			//SAK-11647 - Group-aware dropboxes
+			try
+			{
+				String currentUser = SessionManager.getCurrentSessionUserId();
+				Site site = SiteService.getSite(ToolManager.getCurrentPlacement().getContext());
+				
+				if ((!AuthzGroupService.isAllowed(currentUser,ContentHostingService.AUTH_DROPBOX_MAINTAIN,SiteService.siteReference(site.getId())))
+						&&(AuthzGroupService.isAllowed(currentUser,ContentHostingService.AUTH_DROPBOX_GROUPS,SiteService.siteReference(site.getId()))))
```

Comment:
Rather than duplicating the logic, could you use your new API method `isDropBoxGroups()`?